### PR TITLE
Add/Update test suite

### LINKED ISSUE #5: Improve tests for write_templatized_file
per @ericdatakelly 

## Summary

`write_templatized _file` needs tests.  https://github.com/metrostar/ib_manifest_util/blob/main/ib_manifest_util/util.py#L18

## Acceptance Criteria

- Tests that cover the function run on ci

## Related to

#1 

Looks like we already have a test for this. @kcpevey does the test need more work or should we close this issue?

---
per @kcpevey 

hardening_manifest_path = TEST_DATA_DIR.joinpath("hardening_manifest.yaml") should be output_path.

All the paths and the contents should be moved into conf.py so that they can be reused (potentially as fixtures).

contents is the contents of hardening_manifest.yml so its variable name should reflect that.

### PR DESCRIPTION
**Pull request by kcpevey**
Resolves #5

Changes proposed in this pull request:
- adds test for templatized writers
- adds marks for tests requiring web access
- ensures our test suite won't overwrite sample data
- adds an example script for running update_repo
- adds fixtures for paths to templates and for content to feed into templates.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+markers =
+    web: internet required to run this test (deselect with '-m \"not web\"')
```
Comment (iameskild) [inline pytest.ini:3]: Out of curiosity, for the tests that require an internet connect, would the alternative been to download those files and monkeypatch them?

**Review by iameskild — APPROVED** [review]
Hi @kcpevey, this looks good to me! I ran the tests locally with and without `-m "not web"` set.  

A slightly related question would be are all of the files in the `tests/data` used as inputs for some of these tests? If there are some that we don't need, I guess we could remove them here.

File: pytest.ini (hunk 1 above)
Comment (kcpevey) [reply pytest.ini:3]: yes, a (possibly better) solution would be to monkeypatch the bits that require conda-vendor and also our download functions. For now, its sufficient to let it reach out for the tests while we focus our attention on more important aspect of the codebase.

**Comment by kcpevey** [comment]
>If there are some that we don't need, I guess we could remove them here.

The files are all ones that would be needed when the test suite is fully fleshed out. The plan was to incrementally build those tests. I'd like to leave them as-is awaiting a test.

**Pull request merged**